backend/main.py

```diff
@@ -21,62 +17,46 @@ class StandardResponse(BaseModel):
     data: Any = None
 
 
-if os.path.exists(".env"):
-    load_dotenv(".env")
-
-redis_client = redis.Redis(host="redis", port=6379, db=0)
-
-spotihue = spotihue.SpotiHue(
-    os.environ.get("SPOTIFY_SCOPE"),
-    os.environ.get("SPOTIFY_CLIENT_ID"),
-    os.environ.get("SPOTIFY_CLIENT_SECRET"),
-    os.environ.get("SPOTIFY_REDIRECT_URI"),
-    os.environ.get("HUE_BRIDGE_IP_ADDRESS"),
-    redis_client
-)
-
-celery_app = celery.Celery("celery_app", broker="redis://redis:6379")
 fast_app = FastAPI()
 
 
-@celery_app.task
-def run_spotihue(lights: List[str]) -> None:
-    logger.info("Running spotihue")
+@fast_app.get("/hue-ready")
+def hue_setup_complete():
+    hue_set_up = spotihue.hue_ready()
+    return StandardResponse(success=True, message='Hue setup complete' if hue_set_up else 'Hue setup not complete',
+                            data={'ready': hue_set_up})
 
-    while spotihue.ascertain_track_playing():
-        last_track_info = redis_client.hgetall(constants.REDIS_TRACK_INFORMATION_KEY)
-        last_track_album_artwork_url = last_track_info.get(b"track_album_artwork_url")
 
-        if last_track_album_artwork_url:
-            last_track_album_artwork_url = last_track_album_artwork_url.decode("utf-8")
-
-        track_info = spotihue.retrieve_current_track_information()
-        track_album_artwork_url = track_info["track_album_artwork_url"]
+@fast_app.post("/setup-hue")
+def setup_hue():
+    try:
+        tasks.setup_hue.delay(retries=3)
+    except Exception as e:
+        logger.error(f'Error invoking Hue setup task: {e}')
+        return StandardResponse(success=False, message='Error invoking Hue setup task', data={"setup_running": False})
 
-        redis_client.hset(
-            constants.REDIS_TRACK_INFORMATION_KEY,
-            mapping=track_info
-        )
+    return StandardResponse(success=True, message='Hue setup task running', data={"setup_running": True})
 
-        if last_track_album_artwork_url != track_album_artwork_url:
-            logger.info("Syncing lights")
-            spotihue.sync_lights_music(track_album_artwork_url, lights)
 
-        sleep_duration = random.uniform(2, 4)
-        time.sleep(sleep_duration)
+@fast_app.get("/spotify-ready")
+def spotify_authorization_complete():
+    spotify_authorized = spotihue.spotify_ready()
+    return StandardResponse(success=True, message='Authorized' if spotify_authorized else 'Not Authorized',
+                            data={'ready': spotify_authorized})
 
 
-@fast_app.get("/spotify-ready")
-def spotify_authorized():
-    spotify_auth_manager = spotihue.spotify_oauth
+@fast_app.get("/authorize-spotify")
+def authorize_spotify():
+    spotify_user_auth_url = spotihue.spotify_oauth.auth_url
 
-    spotify_token = spotify_auth_manager.validate_token(
-        spotify_auth_manager.cache_handler.get_cached_token()
-    )
-    token_exists = bool(spotify_token is not None)
+    try:
+        tasks.listen_for_spotify_redirect.delay()
```

Comment:
This is really good. I think this is also something to keep in mind as well:
https://fastapi.tiangolo.com/tutorial/background-tasks/. Didn't know that this existed.

Reply:
Ooh nice, thanks for this  :) didn't know these were a thing! These would totally work too. Since we're already using celery though, I'm ok with leaving the tasks as-is for now if you are...

Reply:
Yeah, definitely leave it! I was just researching.